Flaky test `TestRunWithLogBinary`

### Description

`TestRunWithLogBinary` seems flaky recently.

### Steps to reproduce the issue

Some failed CI jobs:

https://github.com/containerd/nerdctl/actions/runs/3534631174/jobs/5931730823
https://github.com/containerd/nerdctl/actions/runs/3531018664/jobs/5923693580
https://github.com/containerd/nerdctl/actions/runs/3525013201/jobs/5911181827
https://github.com/containerd/nerdctl/actions/runs/3533600557/jobs/5929433503
https://github.com/containerd/nerdctl/actions/runs/3533435718/jobs/5929056734

### Describe the results you received and expected

Test should pass if no related change in a PR.

Current error message:

```shell
=== CONT  TestRunWithLogBinary
    run_test.go:451: assertion failed: res.ExitCode is not exitCode: time="2022-11-23T10:25:30Z" level=fatal msg="container \"ac12cfabc868dfb4ef4d2b9618bbfc9600e26ccc479ea60188e1174045b1bfb9\" in namespace \"nerdctl-test\": not found"
        
--- FAIL: TestRunWithLogBinary (25.00s)
```

```shell
=== CONT  TestRunWithLogBinary
    run_test.go:451: assertion failed: res.ExitCode is not exitCode: time="2022-11-23T16:16:45Z" level=fatal msg="container not created from an image: not found"
        
--- FAIL: TestRunWithLogBinary (31.58s)
```

### What version of nerdctl are you using?

main

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

_No response_

### Host information

_No response_

--- a/cmd/nerdctl/run_test.go
+++ b/cmd/nerdctl/run_test.go
@@ -440,7 +440,7 @@ COPY --from=builder /go/src/logger/logger /
 	base.Cmd("container", "rm", "-f", containerName).AssertOK()
 	base.Cmd("run", "-d", "--log-driver", fmt.Sprintf("binary://%s/logger", tmpDir), "--name", containerName, testutil.CommonImage,
 		"sh", "-euxc", "echo foo; echo bar").AssertOK()
-	defer base.Cmd("container", "rm", "-f", containerName)
+	defer base.Cmd("container", "rm", "-f", containerName).AssertOK()
 
 	inspectedContainer := base.InspectContainer(containerName)
 	bytes, err := os.ReadFile(filepath.Join(os.TempDir(), fmt.Sprintf("%s_%s.log", inspectedContainer.ID, "stdout")))

